src/D2Reader/D2DataReader.cs

```diff
@@ -381,7 +381,7 @@ private D2Unit UnitByTypeAndGuid(D2UnitType type, int guid)
                 // 1.13
                 // for 1.13d see function D2Client.dll+89CE0
                 // for 1.13c see function around D2Client.QueryInterface+FB14
-                var unitAddrPointer = (IntPtr)memory.Units113 + (int)guid * 4;
+                var unitAddrPointer = (IntPtr)memory.Units113 + (int)(guid & 0x7F) * 4;
```

Comment:
`& 0x7f` was missing (note that some lines below, guid with its original value is used (which is correct)